lib/MusicBrainz/Server/Controller/Role/EditRelationships.pm

```diff
@@ -191,8 +191,8 @@ role {
 
                 $args{entity0} = $field->{forward} ? $source : $target;
                 $args{entity1} = $field->{forward} ? $target : $source;
-                $args{entity0_credit} = $field->{entity0_credit};
-                $args{entity1_credit} = $field->{entity1_credit};
+                $args{entity0_credit} = $field->{entity0_credit} if exists $field->{entity0_credit};
```

Comment:
The changes to generic.js make it so that these fields are only submitted if they're changed. So we don't want to "update" any data for fields that don't exist.